stakedb/ticketpool.go

```diff
@@ -603,7 +603,7 @@ func (tp *TicketPool) AdvanceToTip() (int64, error) {
 	tp.mtx.Lock()
 	defer tp.mtx.Unlock()
 	err := tp.advanceTo(tp.tip)
-	return tp.cursor - 1, err
+	return tp.cursor, err
```

Comment:
Previously, by returning the `cursor - 1`, this was giving an index of the last element in the `diffs` slice.  However, this was assumed to be height in several places including `NewStakeDatabase` and `LoadAndRecover`.  By returning `tp.cursor`, this will now correspond to height.

Reply:
Tested with fresh DB startup as well as resume.